frontend/app/routes/payment-confirmation.tsx

```diff
@@ -8,21 +15,58 @@ export const meta: MetaFunction = () => {
   ]
 }
 
+export async function loader({ request, context }: LoaderFunctionArgs) {
+  const { env } = context.cloudflare
+  const url = new URL(request.url)
+  const params: { [key: string]: string } = {}
+
+  url.searchParams.forEach((value, key) => {
+    params[key] = value
+  })
```

Comment:
Can do something like:
```suggestion
  const params = Object.fromEntries([...url.searchParams])
```